File: src/shared/appIdentity.ts
import fs from "node:fs";
import os from "node:os";
import path from "node:path";

export const APP_NAME = "Jarvey";

export function resolveApplicationSupportRoot(): string {
  const baseDirectory = path.join(os.homedir(), "Library", "Application Support");
  return path.join(baseDirectory, APP_NAME);
}

export function resolveAppLogsDirectory(): string {
  const logsDirectory = path.join(resolveApplicationSupportRoot(), "logs");
  fs.mkdirSync(logsDirectory, { recursive: true });
  return logsDirectory;
}

File: Sources/JarveyNative/AppIdentity.swift
import Foundation

enum AppIdentity {
  static let appName = "Jarvey"
  static let voiceMessageHandlerName = "jarveyVoice"
  static let voiceBridgeObjectName = "jarveyVoiceBridge"

  static func applicationSupportRoot(fileManager: FileManager = .default) -> URL {
    let baseDirectory =
      fileManager.urls(for: .applicationSupportDirectory, in: .userDomainMask).first
      ?? URL(fileURLWithPath: NSHomeDirectory())
        .appending(path: "Library", directoryHint: .isDirectory)
        .appending(path: "Application Support", directoryHint: .isDirectory)
    return baseDirectory.appending(path: appName, directoryHint: .isDirectory)
  }

  static func logsDirectory(fileManager: FileManager = .default) -> URL {
    let logsDirectory = applicationSupportRoot(fileManager: fileManager)
      .appending(path: "logs", directoryHint: .isDirectory)
    try? fileManager.createDirectory(at: logsDirectory, withIntermediateDirectories: true)
    return logsDirectory
  }
}
